Training24/test.c

```diff
@@ -0,0 +1,170 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// test.c
+// Program on A3 branch.
+// Convert the given decimal number to different forms (Decimal, Hexadecimal, Binary)
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+#include "game.h"
+
+// ANSI escape codes for colors
+#define COLOR_RESET   "\033[0m"
+#define COLOR_GREEN   "\033[32m"
+#define COLOR_RED     "\033[31m"
+#define COLOR_YELLOW  "\033[33m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");              // Windows-specific command to clear the screen
+#else
+   system ("clear");           // Unix/Linux/MacOS-specific command to clear the screen
+#endif
+}
+
+/// <summary>Function to print test results for binary conversion.</summary>
+void FormatBin (int numTests, int inputs[], const char* expected_results[], const char* actual_results[], int pass[]) {
+   // Adjust column widths for alignment
+   printf (COLOR_YELLOW "-------------------------------------------- Test Results for Binary ------------------------------------------\n" COLOR_RESET);
+   printf ("| Test Case | Input       | Expected                            | Actual                               |Result|\n");
+   printf ("|-----------|-------------|-------------------------------------|--------------------------------------|------|\n");
+   for (int i = 0; i < numTests; i++) {
+      // Adjust column widths to match header alignment
+      printf ("| %-9d | %-11d | %-35s |" COLOR_GREEN " % -36s "COLOR_RESET"| % -6s | \n", i + 1, inputs[i], expected_results[i], actual_results[i] ? actual_results[i] : "NULL",
+         pass[i] ? COLOR_GREEN "PASS" COLOR_RESET : COLOR_RED "FAIL" COLOR_RESET);
+   }
+   printf ("\n");
+}
+
+/// <summary>Function to print test results for Hexadecimal conversion.</summary>
+void FormatHex (int numTests, int inputs[], const char* expected_results[], const char* actual_results[], int pass[]) {

```

Comment:
FormatHex and FormatBin peform the same task in exactly the same way. They do  not need to be two different functions. 
See [Code Re-use](https://en.wikipedia.org/wiki/Code_reuse).